httpcore/_async/http2.py

```diff
@@ -314,17 +314,41 @@ async def send_headers(
     ) -> None:
         scheme, hostname, port, path = url
         default_port = {b"http": 80, b"https": 443}.get(scheme)
-        if port is None or port == default_port:
+
+        # In HTTP/2 the ':authority' pseudo-header is used instead of 'Host'.
+        # For convenience we automatically treat any 'Host' header value
+        # as a value for ':authority', allowing users to pass 'Host' whenever they
+        # want to customize the host header regardless of the HTTP version in use.
+        authority = None
+        for k, v in headers:
+            if k == b"host":
+                authority = v
+                break
+
+        if not authority:
```

Comment:
Let's try to stick to `is None` / `is not None` checks as closely as we can; in the past we've had issues related to eg passing or not passing empty values or zeroes, so checking specifically for None if that's the intent should be preferable.

```suggestion
        if authority is None:
```